theano/tensor/elemwise.py

```diff
@@ -1993,10 +1999,24 @@ class Sum(CAReduceDtype):
 
     """
 
+    __props__ = ("axis", "dtype", "acc_dtype")
+
     def __init__(self, axis=None, dtype=None, acc_dtype=None):
         CAReduceDtype.__init__(self, scalar.add, axis=axis,
                                dtype=dtype, acc_dtype=acc_dtype)
 
+    def __str__(self):
+        name = self.__class__.__name__
+        axis = ""
+        if self.axis is not None:
+            axis = ", ".join(str(x) for x in self.axis)
+            axis = "axis=[%s], " % axis
+        return "%s{%sacc_dtype=%s}" % (
+            name,
+            axis,
+            str(self.acc_dtype)
+        )
```

Comment:
I'm slightly confused here and to test i added this method. When I added `__props__` to this Op, there were errors in test/test_printing as the string that this Op returns doesn't match with the ground truth. I then realized, it is creating the `__str__` method from the `__props__` inside the utils instead of it using the parent class's `__str__`. that's why I added this method, similar to `CAReduceDtype`. 
My confusion here is, even if i add `__props__` shouldn't the parent's class `__str__` method be used instead of creating one on it's own ? Being more precise, shouldn't line 199 in theano/gof/utils.py evaluate to `False` ?